src/App.tsx

```diff
@@ -1,24 +1,149 @@
-import React, { useState } from 'react';
+import React, { useReducer } from 'react';
 import { StyleSheet, Text, View } from 'react-native';
 import AddItem from './components/AddItem';
 import AddPartyMember from './components/AddPartyMember';
 import Tax from './components/Tax';
 import Tip from './components/Tip';
 
+enum Actions {
+  ADD_ITEM = 'add_item',
+  ADD_PARTY_MEMBER = 'add_party_member',
+  SET_TIP = 'set_tip',
+  SET_TAX = 'set_tax',
+}
+
+type AppStateType = {
+  total: number;
+  items: {
+    [name: string]: {
+      price: number;
+      purchaserList: string[];
+    };
+  };
+  party: string[];
+  tip: number;
+  tax: number;
+};
+
+type PartyMemberDetailsType = {
+  name: string;
+};
+
+type ItemDetailsType = {
+  name: string;
+  price: number;
+  purchaserList: string[];
+};
+
+type TaxDetailsType = {
+  amount: number;
+};
+
+type TipDetails = {
+  amount: number;
+};
+
+type ActionArgs = {
+  type: Actions;
+  partyMemberDetails: PartyMemberDetailsType;
+  itemDetails: ItemDetailsType;
```

Comment:
Can we rename these to ItemDetails, TaxDetails, TipDetails instead of ItemDetailsType, etc...? They don't really describe a "type" of XX-Details. Or ItemDetailsInstance, if we don't want to repeat: `itemDetails: ItemDetails` but I feel like we don't need to append Instance or another word.

Reply:
done! i also renamed `AppStateType` to just `AppState`